docs/en/FAQ/Load-outside-jar-to-classpath.md

```diff
@@ -0,0 +1,24 @@
+# Load outside jar to classpath 
+ 
+### Problem
+In some cases, especially when the target java app is packaged in an old way,that mean the depency libs directory is out of the target application jar. We need to add some VM options to make sure the java applications starts well, like "java -cp **/lib -ext.dirs **/lib" .
+But when we add these VM options, the target ClassLoaders may not be able to load any classes from skywalking-agent.jar cause it's out of the path i have defined previously. The target app classloader or any of its parent classloader will load the class with failure and push it down to AgentClassLoader to load the class ,and finally ,the not class found exception will stop the application from start.
+
+### Resolve
+"-Xbootclasspath/a:" can solve this problem.
+By the way ,we can use the BootstrapClassLoader to load the them.
```

Comment:
Use `BootstrapClassLoader`? Is this working? I remember all agent codes(except few JDK instrumentation) are loaded by Application ClassLoader only, please confirm.

Reply:
Yes, that will use BootstrapClassLoader. I have a question : Why the target ClassLoader(s) need to load classes from skywalking-agent.jar? I think I got it wrong.

Reply:
I think we don't need `BootstrapClassLoader`, we need application's codes could access agent's classes, which are loaded by application class loader due to `-javaagent:agent.jar`.
That is why I am confused about your writing.

Reply:
Got it. What if we can get application's codes and  agent's codes to load  by application class loader.  For example: Instrumentation.**_appendToSystemClassLoaderSearch_**(application's  jar or application's dependency jar) ? Is this working?

Reply:
I am not sure. But basically, I don't recommend you to manipulate the class loader of users' codes. It is hard to tell whether they are using some private class loader mechanism, which could be failure due to this change.
This kind of things could work in your own practice, but it may not work for anyone else.

Reply:
I see.So I close this PR ?

Reply:
Agree, I think we need some other ideas to resolve this issue. :)